mango-python/bdgenomics/mango/variantViz.py

```diff
@@ -0,0 +1,192 @@
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        Initializes HetHomRatioDistribution class.
+        Retrieves the call rate and missing rate for each sample in a genotypeRDD
+
+        Args:
+            :param ss: SparkContext
+            :param genotypeRDD: bdgenomics genotypeRDD
+        """
+
+        df =  genotypeRDD.toDF()
+        data_het =  df.where( (df.alleles[0] == u'REF') & (df.alleles[1] == u'ALT')).groupBy("sampleid").count().collect()
+        self.hets =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in data_het }
+        data_hom =  df.where( (df.alleles[0] == u'ALT') & (df.alleles[1] == u'ALT')).groupBy("sampleid").count().collect()
+        self.homs =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in data_hom }
+
+    def plot(self):
+        """
+        Computes and returns a list of per sample Het/Hom ratios suitable for plotting
+
+        Returns:
+          a list containing the Het/Hom ratio statistic for each sample in the dataset
+        """
+        hetHomRatio = [ float(self.hets[sampleid])/float(self.homs[sampleid]) for sampleid in self.homs.keys() ]
+        return hetHomRatio
+
+
+class GenotypeCallRatesDistribution(object):
+    """ GenotypeCallRatesDistribution class.
+    GenotypeCallRatesDistribution computes a distribution of per-sample genotype call rates from a genotypeRDD.
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        Initializes a GenotypeCallRatesDistribution class.
+        Retrieves counts of called and missing genotypes from a genotypeRDD.
+
+        Args:
+            :param ss: SparkContext
+            :param genotypeRDD: bdgenomics genotypeRDD
+        """
+
+        df =  genotypeRDD.toDF()
+        called_df =  df.where( (df.alleles[0] != u'NO_CALL')).groupBy("sampleid").count().collect()
+        self.called_map =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in called_df }
+
+        missing_df =  df.where( (df.alleles[0] == u'NO_CALL')).groupBy("sampleid").count().collect()
+        self.missing_map =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in missing_df }
+
+    def plot(self):
+        """
+        Computes and returns a list of per sample call rates suitable for plotting.
+
+        Returns:
+           A list of per sample call rates.
+        """
+        callRatePerSample = [ float(self.called_map[sampleid])/ \
+                              ( float(self.missing_map[sampleid])+float(self.called_map[sampleid])) \
+                              for sampleid in self.called_map.keys() ]
+
+        return callRatePerSample
+
+
+class QualityByDepthDistribution(object):
+    """ QualityByDepthDistribution class.
+    QualityByDepthDistribution calculates the Quality By Depth (QD) score distribution from a variantRDD.
+    """
+    def __init__(self,ss,variantRDD):
```

Comment:
Any distribution code should extend CountDistribution if it can (see https://github.com/bigdatagenomics/mango/blob/master/mango-python/bdgenomics/mango/coverage.py#L36)